src/espIdf/hints/index.ts

```diff
@@ -0,0 +1,153 @@
+import * as os from "os";
+import * as yaml from "js-yaml";
+import * as fs from "fs";
+import * as idfConf from "../../idfConfiguration";
+import * as vscode from "vscode";
+
+class ReHintPair {
+  re: string;
+  hint: string;
+  match_to_output: boolean;
+
+  constructor(re: string, hint: string, match_to_output: boolean = false) {
+    this.re = re;
+    this.hint = hint;
+    this.match_to_output = match_to_output;
+  }
+}
+
+class ErrorHint {
+  public type: "error" | "hint";
+  public children: ErrorHint[] = [];
+
+  constructor(public label: string, type: "error" | "hint") {
+    this.type = type;
+  }
+
+  addChild(child: ErrorHint) {
+    this.children.push(child);
+  }
+}
+
+export class ErrorHintProvider implements vscode.TreeDataProvider<ErrorHint> {
+  constructor(private context: vscode.ExtensionContext) {}
+  private _onDidChangeTreeData: vscode.EventEmitter<
+    ErrorHint | undefined | null | void
+  > = new vscode.EventEmitter<ErrorHint | undefined | null | void>();
+  readonly onDidChangeTreeData: vscode.Event<
+    ErrorHint | undefined | null | void
+  > = this._onDidChangeTreeData.event;
+
+  private data: ErrorHint[] = [];
+
+  searchError(errorMsg: string, workspace) {
+    const espIdfPath = idfConf.readParameter(
+      "idf.espIdfPath",
+      workspace
+    ) as string;
+    const hintsPath = getHintsYmlPath(espIdfPath);
+    const fileContents = fs.readFileSync(hintsPath, "utf-8");
```

Comment:
Maybe we can add some file exist validation here ? It could fail if the file doesn't exists. Also which versions of ESP-IDF support hints ? Maybe a version validation would be good to know.

Could we replace readFileSync with a Promise or async version like the one from fs-extra ?